File: web/blog/media/prototype_one/data/player.js
data.push( { type : 'file', name : 'player.h', content : [






	{ line : 8, type : 'class', name : 'Player', content : {
		public : [
			{ line : 10, type : 'method', name : 'Player', params : [ 'Field::Token', 'std::string' ], content : [
				{ line : 11, type : 'use', member : 'token_' },
				{ line : 12, type : 'use', member : 'name_' }
			]},

			{ line : 15, type : 'method', name : 'Turn', abstraction : 'pure_virtual', return : 'Field::Move', params : [ 'Field' ] },

			{ line : 17, type : 'member', name : 'token_', class : 'Field::Token' },
			{ line : 18, type : 'member', name : 'name_', class : 'std::string' },
		]}
	}
]});

File: bin/app/user/projects/tictactoe_cpp/src/player.cpp
#include "player.h"

Player::Player( Field::Token token, const char* name )
	: token_( token )
	, name_( name ) {
}

Player::~Player() {
}

const Field::Token& Player::getToken() const
{
	return token_;
}

const char* Player::getName() const
{
	return name_;
}
